tests/api_connexion/endpoints/test_dag_endpoint.py

```diff
@@ -14,35 +14,109 @@
 # KIND, either express or implied.  See the License for the
 # specific language governing permissions and limitations
 # under the License.
+import os
 import unittest
+from datetime import datetime
 
 import pytest
 
+from airflow import DAG
+from airflow.models import DagBag
+from airflow.models.serialized_dag import SerializedDagModel
+from airflow.operators.dummy_operator import DummyOperator
 from airflow.www import app
+from tests.test_utils.db import clear_db_dags, clear_db_runs, clear_db_serialized_dags
 
 
 class TestDagEndpoint(unittest.TestCase):
+    dag_id = "test_dag"
+    task_id = "op1"
+
+    @staticmethod
+    def clean_db():
+        clear_db_runs()
+        clear_db_dags()
+        clear_db_serialized_dags()
+
     @classmethod
     def setUpClass(cls) -> None:
         super().setUpClass()
         cls.app = app.create_app(testing=True)  # type:ignore
+        cls.app_serialized = app.create_app(testing=True)  # type:ignore
+
+        with DAG(
+            cls.dag_id, start_date=datetime(2020, 6, 15), doc_md="details"
+        ) as dag:
+            DummyOperator(task_id=cls.task_id)
+
+        cls.dag = dag  # type:ignore
+
+        dag_bag = DagBag(os.devnull, include_examples=False)
+        dag_bag.dags = {dag.dag_id: dag}
+        cls.app.dag_bag = dag_bag  # type:ignore
+
+        dag_bag = DagBag(os.devnull, include_examples=False, store_serialized_dags=True)
+        cls.app_serialized.dag_bag = dag_bag  # type:ignore
 
     def setUp(self) -> None:
+        self.clean_db()
         self.client = self.app.test_client()  # type:ignore
+        self.client_serialized = self.app_serialized.test_client()  # type:ignore
```

Comment:
I am not sure if this is the right way but nothing else comes to my mind. What we have here are two testing clients one with single DAG dagbag and second one with empty one with serialization switched on

Reply:
Maybe just give up this base class and divide the whole into two classes?

Reply:
I was thinking about creating a new app in test with serialization, WDYT?